middleware/compress.go

```diff
@@ -144,3 +156,32 @@ func gzipCompressPool(config GzipConfig) sync.Pool {
 		},
 	}
 }
+
+// removeOneHeaderValue manipulates the []string value of one header to
+// remove an unwanted item from it.
+func removeOneHeaderValue(header http.Header, key, unwanted string) {
+	key = textproto.CanonicalMIMEHeaderKey(key)
+	values := header[key]
+	for i, v := range values {
+		if v == unwanted {
+			values = append(values[:i], values[i+1:]...)
+			header[key] = values
+			return
+		}
+	}
+}
+
+// gzipAccepted determines whether the client will accept gzip-encoded responses.
+// This is only an approximation: we don't handle special cases such as "gzip;q=0"
+// that are allowed by IETF RFC.
+func gzipAccepted(header http.Header) bool {
```

Comment:
why not let it be as it was `strings.Contains(c.Request().Header.Get(echo.HeaderAcceptEncoding), gzipScheme)`?

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Encoding refers to 
https://www.iana.org/assignments/http-parameters/http-parameters.xml#http-parameters-1  which list "official" variants of values. Googling seems to say that `pack200-gzip` seems to be gzip so it should be safe just to check if header contains that part.

Reply:
Please restore old `strings.Contains(c.Request().Header.Get(echo.HeaderAcceptEncoding), gzipScheme)`. This kind of check does not add value, introduces unnecessary complexity. And is not correct for cases when header value contains `x-gzip` which is valid alias (although deprecated) for gzip by https://www.iana.org/assignments/http-parameters/http-parameters.xml#http-parameters-1

`strings.Contains` is good enough check.